tests/test_matrix.py

```diff
@@ -38,9 +90,98 @@ def test_crout_matrix_decomposition(self):
                            [0, 0, 1.0, 0.0],
                            [0, 0, 0, 1.0]]),
                          crout_matrix_decomposition.crout_matrix_decomposition(
-                             [[2,1,3,1], [4,1,4,1], [6,1,7,1], [8,1,9,1]]))
-                           
-    
-        
+                             [[2, 1, 3, 1], [4, 1, 4, 1], [6, 1, 7, 1], [8, 1, 9, 1]]))
+
+
+class TestMultiply(unittest.TestCase):
+    """[summary]
+    Test for the file multiply.py
+
+    Arguments:
+        unittest {[type]} -- [description]
+    """
+
+    def test_multiply(self):
+        self.assertEqual(multiply.multiply(
+            [[1, 2, 3], [2, 1, 1]], [[1], [2], [3]]), [[14], [7]])
+
+
+class TestRotateImage(unittest.TestCase):
+    """[summary]
+    Test for the file rotate_image.py
+
+    Arguments:
+        unittest {[type]} -- [description]
+    """
+
+    def test_rotate_image(self):
+        self.assertEqual(rotate_image.rotate(
+            [[1, 2, 3], [4, 5, 6], [7, 8, 9]]), [[7, 4, 1], [8, 5, 2], [9, 6, 3]])
+
+
+class TestSparseDotVector(unittest.TestCase):
+    """[summary]
+    Test for the file sparse_dot_vector.py
+
+    Arguments:
+        unittest {[type]} -- [description]
+    """
+
+    def test_sparse_dot_vector(self):
+        self.assertEqual(sparse_dot_vector.dot_product(sparse_dot_vector.vector_to_index_value_list(
+            [1., 2., 3.]), sparse_dot_vector.vector_to_index_value_list([0., 2., 2.])), 10)
+
+
+class TestSpiralTraversal(unittest.TestCase):
+    """[summary]
+    Test for the file spiral_traversal.py
+
+    Arguments:
+        unittest {[type]} -- [description]
+    """
+
+    def test_spiral_traversal(self):
+        self.assertEqual(spiral_traversal.spiral_traversal(
+            [[1, 2, 3], [4, 5, 6], [7, 8, 9]]), [1, 2, 3, 6, 9, 8, 7, 4, 5])
+
+
+class TestSudokuValidator(unittest.TestCase):
+    """[summary]
+    Test for the file sudoku_validator.py
+
+    Arguments:
+        unittest {[type]} -- [description]
+    """
+
+    def test_sudoku_validator(self):
+        self.assertTrue(
+            sudoku_validator.valid_solution(
+                [
+                    [
+                        5, 3, 4, 6, 7, 8, 9, 1, 2], [
+                        6, 7, 2, 1, 9, 5, 3, 4, 8], [
+                        1, 9, 8, 3, 4, 2, 5, 6, 7], [
+                            8, 5, 9, 7, 6, 1, 4, 2, 3], [
+                                4, 2, 6, 8, 5, 3, 7, 9, 1], [
+                                    7, 1, 3, 9, 2, 4, 8, 5, 6], [
+                                        9, 6, 1, 5, 3, 7, 2, 8, 4], [
+                                            2, 8, 7, 4, 1, 9, 6, 3, 5], [
+                                                3, 4, 5, 2, 8, 6, 1, 7, 9]]))
```

Comment:
can you align the columns?

Reply:
Sure. It's auto-formatted by the autopep8 LOL